python/requirements.txt

```diff
@@ -24,7 +24,7 @@ grpcio == 1.54.2; sys_platform == "darwin"
 grpcio >= 1.54.2; sys_platform != "darwin"
 numpy>=1.20
 
-pyarrow >= 6.0.1
+pyarrow==14.0.2
```

Comment:
i think we still want to support pyarrow 6, @bveeramani , @scottjlee for input

Reply:
Yeah, that's right.

Reply:
I'm having difficulty solving the version issue because it seems like there are lot of dependencies between versions. For example, here is a crash I just saw:
```
[2024-03-27T19:00:29Z] #9 125.5 The conflict is caused by:
[2024-03-27T19:00:29Z] #9 125.5     The user requested pyarrow==14.*
[2024-03-27T19:00:29Z] #9 125.5     pymongoarrow 0.5.1 depends on pyarrow<9.1 and >=9
[2024-03-27T19:00:29Z] #9 125.5     The user requested pyarrow==14.*
[2024-03-27T19:00:29Z] #9 125.5     pymongoarrow 0.5.0 depends on pyarrow<9.1 and >=9
[2024-03-27T19:00:29Z] #9 125.5     The user requested (constraint) pyarrow==14.0.2
```
Do you have any suggestion for how to resolve this?

Reply:
I think if you want to pin pyarrow for the ci test environment only, you can add a line to this file https://github.com/ray-project/ray/blob/master/python/requirements/test-requirements.txt like this

```
pyarrow==14.0.2; sys_platform == "darwin"
```

and then keep this python/requirements.txt unchanged. This means that users can still install any pyarrow (>=6) version that they want on macos; under the risk that these test scenarios might or might not be flaky (as you said, those tests are not really flaky on your machine).

Reply:
oh you might also need to change the one in requirements.txt into 

`
pyarrow >= 6.0.1; sys_platform != "darwin"
pyarrow >= 6.0.1; sys_platform == "darwin"
`

a little bit dump but it's what it is

Reply:
Dumb question -- if the motivation for this PR is to deflake tests, would it make sense to just update the dependencies on CI and not change the underlying Ray requirements?

Reply:
what you said yes ;)